agent/engine/docker_task_engine.go

```diff
@@ -1312,43 +1299,27 @@ func (engine *DockerTaskEngine) buildCNIConfigFromTaskContainer(
 	return cniConfig, nil
 }
 
-func (engine *DockerTaskEngine) inspectContainerByName(taskArn, containerName string) (*types.ContainerJSON, error) {
-	containers, ok := engine.state.ContainerMapByArn(taskArn)
-	if !ok {
-		return nil, errors.New("engine: failed to find the pause container, no containers in the task")
-	}
-
-	pauseContainer, ok := containers[containerName]
-	if !ok {
-		return nil, errors.New("engine: failed to find the pause container")
+func (engine *DockerTaskEngine) inspectContainer(taskArn, container *apicontainer.Container) (*types.ContainerJSON, error) {
+	dockerID, err := engine.getDockerID(task, container)
```

Comment:
Yes it's true, getDockerID could sometimes return the name, I'm not sure how to call the function to indicate that.

You are correct that this would change to mostly calling the docker API using the docker runtime ID. The docker API for the most part is very liberal in what it accepts in terms of this. You can pass in the name, full runtime id, short name, or short runtime ID. That is part of why I wanted to make this change, so that it is more standardized on using a single type of "docker id".

Reply:
for example, previous to this PR we were called RemoveContainer using DockerName, but if you look at the implementation it is referred to as DockerID in our code, and then ultimately containerID in the docker client codebase: https://github.com/aws/amazon-ecs-agent/blob/1ebf0604c13013596cfd4eb239574a85890b13e8/agent/dockerclient/dockerapi/docker_client.go#L701

Reply:
I see. I'd prefer to either only use DockerConainer everywhere or to add these attributes directly to the container struct. But that's a much larger change. I guess this is a good iteration